source/set-required-deep.d.ts

```diff
@@ -0,0 +1,45 @@
+import type {OmitDeep} from './omit-deep';
+import type {Paths} from './paths';
+import type {PickDeep} from './pick-deep';
+import type {RequiredDeep} from './required-deep';
+import type {SimplifyDeep} from './simplify-deep';
+
+/**
+Create a type that makes the given keys required. You can specifiy deeply nested key paths. The remaining keys are kept as is.
+
+Use-case: You want to define as required only a some nested keys inside a model.
+
+@example
+```
+import type {SetRequiredDeep} from 'type-fest';
+
+type Foo = {
+	a?: number;
+	b?: string;
+	array?: {
+		c?: number
+	}[]
+}
+
+type SomeRequiredDeep = SetRequiredDeep<Foo, 'a' | 'array.c'>;
+// type SomeRequiredDeep = {
+// 	a: number; // Is now required
+// 	b?: string;
+// 	array: {
+//    c: number // Is now required
+//  }[]
+// }
+```
+
+@category Object
+*/
+export type SetRequiredDeep<BaseType, KeyPaths extends Paths<BaseType>> =
```

Comment:
I'm playing with this solution using the following example:

```ts
type Teacher = { address?: { postcode?: string; city?: string } };
type P1 = SetRequiredDeep<Teacher, 'address' | 'address.city'>;
//      ^?
```

This gives me:

```ts
type P1 = {
    address: {
        postcode: string;
        city: string;
    };
} | {
    address: {
        postcode: string;
        city: string;
    };
}
```

Is it expected?

Reply:
I agree

Reply:
Yes I think my current implementation could not work for this, since I use `RequiredDeep` and `RequiredDeep` makes all child keys of an object required as well.